packages/theme/src/cli/utilities/theme-environment/storefront-renderer.ts

```diff
@@ -136,3 +136,11 @@ function themeAccessHeaders(session: DevServerSession) {
     'X-Shopify-Access-Token': session.token,
   }
 }
+
+function cleanHeader(headers: {[key: string]: string}): {[key: string]: string} {
+  // Avoid ambiguity in case the headers object contains the
+  // 'Cookie' and 'cookie' keys.
+  delete headers.cookie
+  delete headers.authorization
+  return headers
+}
```

Comment:
Should we start using `new Headers(...)` to avoid problems with case sensitivity? Or would that be harder to use?

Reply:
Thank you for the review, @frandiox! :D

I've considered the `Headers` approach (which would indeed be a bit more elegant). However, this would expose our logic to a proxy that might set both cookies, as we can see in the example below:

```
  const header = new Headers({
    Cookie: 'value1',
    cookie: 'value2',
  });

console.log('>>', header.get('cookie'));
# => value1; value2
```

I've updated the comment to clarify that we want to force the usage of the `Cookie` key and ignore the `cookie` key.